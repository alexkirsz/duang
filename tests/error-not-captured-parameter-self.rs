// this test check the error message when parameters are not all Captured.
use duang::duang;

struct Unit();

impl Unit {
  duang!(
  pub fn foo<T>(&self, a: T,b: f64 = 13.0, c: T = a*a) -> (T,f64,T)
  where
    T: std::ops::Mul<T, Output = T>,
    T: std::fmt::Display,
    T: Copy,
  {
    (a,b,c)
  }
  );
}

fn main() {
}
