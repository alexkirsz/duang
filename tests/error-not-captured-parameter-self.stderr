error: Associated function is not supported yet.
 --> $DIR/error-not-captured-parameter-self.rs:8:17
  |
8 |   pub fn foo<T>(&self, a: T,b: f64 = 13.0, c: T = a*a) -> (T,f64,T)
  |                 ^
